core-common/src/main/java/org/mapstruct/factory/Mappers.java

```diff
@@ -73,6 +75,16 @@ private Mappers() {
                 classLoader = Mappers.class.getClassLoader();
             }
 
+            ServiceLoader<T> loader = ServiceLoader.load( clazz, classLoader );
```

Comment:
It'd be better to do it as a fall-back only after the normal approach, as I suppose using the service loader is slower.